src/GPUActivityAgent.cpp

```diff
@@ -114,8 +161,16 @@ namespace geopm
         // value provided by the policy may not be valid.  In this case approximating
         // f_efficient as midway between F_min and F_max is reasonable.
         if (std::isnan(in_policy[M_POLICY_GPU_FREQ_EFFICIENT])) {
-            in_policy[M_POLICY_GPU_FREQ_EFFICIENT] = (in_policy[M_POLICY_GPU_FREQ_MAX]
-                                                      + gpu_min_freq) / 2;
+            auto all_names = m_platform_io.signal_names();
+            std::string fe_sig_name = "LEVELZERO::GPU_CORE_FREQUENCY_EFFICIENT";
+            if (all_names.count(fe_sig_name) != 0) {
+                in_policy[M_POLICY_GPU_FREQ_EFFICIENT] = m_platform_io.read_signal(fe_sig_name,
+                                                                                   GEOPM_DOMAIN_BOARD, 0);
```

Comment:
I'm guessing you've already run some experiments and found that taking the average Fe value is sufficient/good enough. I realize you've been managing a single/global Fe value all along, but just now it occurred to me to ask of there's any value in tracking it per GPU or "domain_idx".

Reply:
There is some value in tracking Fe per domain_idx, but until recently (#2635 & #2435) there was no convenient method to provide this.
The value of per domain Fe is largely based upon how much variability is exhibited in the Fe values across the domain.